OpenRA.Game/Renderer.cs

```diff
@@ -84,6 +85,7 @@ static IGraphicsDevice CreateDevice(Assembly rendererDll, int width, int height,
 			foreach (RendererAttribute r in rendererDll.GetCustomAttributes(typeof(RendererAttribute), false))
 			{
 				var factory = (IDeviceFactory)r.Type.GetConstructor(Type.EmptyTypes).Invoke(null);
+				Sound.Engine = factory.Initialize();
```

Comment:
Tacking the audio device initialization on as a side-effect is pretty horrible.  You could either:

1.  Change this method to `static Pair<IGraphicsDevice, IAudioDevice> CreateDevices`
2.  Make it return void and set both Sound.Engine and Renderer.Device here
3.  Add a similar `CreateDevice` method to Sound and avoid mixing renderer and audio code.

My preference is strongly towards 3 (not sure why I even mention the first two, but i've already written them now...).